src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy

```diff
@@ -0,0 +1,142 @@
+
+import life.qbic.datamodel.datasets.datastructure.files.DataFile
+import life.qbic.datamodel.datasets.datastructure.folders.DataFolder
+import life.qbic.datamodel.datasets.datastructure.folders.nanopore.*
+
+/**
+ * A dataset that represents a Oxford Nanopore Measurement.
+ *
+ * @author: Sven Fillinger
+ */
+final class OxfordNanoporeMeasurement {
+
+    private final Map<String, DataFolder> folders
+
+    private final List<DataFile> logFiles
+
+    private final MeasurementFolder measurementFolder
+
+    private final boolean pooledMeasurement
+
+    protected OxfordNanoporeMeasurement(String name, String path, List children) {
+        logFiles = new ArrayList<>()
+        folders = new HashMap<>()
+
+        this.measurementFolder = MeasurementFolder.create(name, path, children)
+
+        createContent()
+
+        pooledMeasurement = folders["fast5passed"] ? folders["fast5passed"].getTheChildren().get(0) instanceof Fast5Folder : false
+    }
+
+    static OxfordNanoporeMeasurement create(String name, String path, List children) {
+        return new OxfordNanoporeMeasurement(name, path, children)
+    }
+
+    private void createContent() {
+        measurementFolder.getTheChildren().each { element ->
+            switch (element) {
+                case Fast5PassFolder:
+                    folders["fast5passed"] = element as Fast5PassFolder
+                    break
+                case Fast5FailFolder:
+                    folders["fast5failed"] = element as Fast5FailFolder
+                    break
+                case FastQPassFolder:
+                    folders["fastqpass"] = element as FastQPassFolder
+                    break
+                case FastQFailFolder:
+                    folders["fastqfail"] = element as FastQFailFolder
+                    break
+                case DataFile:
+                    logFiles.add(element as DataFile)
+            }
+        }
+    }
+
+    /**
+     * This method aggregates all fast5 files and fastq files of an Oxford Nanopore
+     * measurement by sample id.
+     *
+     * The resulting data-structure follows this map schema:
+     *
+     * "QBiC sample id":
+     *      "fast5fail": DataFolder
+     *      "fast5pass": DataFolder
+     *      "fastqfail": DataFolder
+     *      "fastqpass": DataFolder
+     * "Other sample id":   // In case of pooled samples
+     *      ...
+     * @return Map A nested map with sample ids and containing data folders
+     */
+    Map<String, Map<String, DataFolder>> getRawDataPerSample(Experiment experiment) {
+        if (pooledMeasurement) {
+            return prepareRawDataFromPooledMeasurement()
+        } else {
+            return prepareRawData(experiment.sampleId)
+        }
+    }
+
+    private Map<String, Map<String, DataFolder>> prepareRawDataFromPooledMeasurement() {
+        final def result = new HashMap()
+        final def pooledSampleIds = folders
+                .get("fast5failed")
+                .getTheChildren()
```

Comment:
```suggestion
                .getChildren()
```
I would suggest to refactor. Should work now.

Reply:
This can cause a lot of confusion in the future since the default getter method is suggested by most IDEs. Groovy generates a default getter that would need to be overwritten or made private to prevent access to it.